core/lib/spree/core/product_duplicator.rb

```diff
@@ -70,5 +70,12 @@ def reset_properties
         end
       end
     end
+
+    def generate_new_sku(sku)
+      loop do
+        sku = "COPY OF #{sku}"
```

Comment:
I don't understand the point of the loop if you are always using the same sku shouldn't this iterate over an index or use a unique timestamp or something.